src/prescriptions/prescription-details.component.tsx

```diff
@@ -37,24 +50,55 @@ const PrescriptionDetails: React.FC<{
   }, [totalAllergies]);
 
   const generateStatusTag: Function = (
-    medicationRequest: MedicationRequest
+    medicationRequest: MedicationRequest,
+    medicationDispenses: Array<MedicationDispense>
   ) => {
-    const status = computeStatus(
+    const medicationRequestStatus = computeMedicationRequestStatus(
       medicationRequest,
       config.medicationRequestExpirationPeriodInDays
     );
-    if (status === "cancelled") {
+    const medicationDispenseStatus =
+      getMostRecentMedicationDispenseStatus(medicationDispenses);
+    const combinedStatus: MedicationRequestMedicationDispenseCombinedStatus =
+      computeMedicationRequestMedicationDispenseCombinedStatus(
+        medicationRequestStatus,
+        medicationDispenseStatus
+      );
+    if (
+      combinedStatus ===
+      MedicationRequestMedicationDispenseCombinedStatus.cancelled
+    ) {
       return <Tag type="red">{t("cancelled", "Cancelled")}</Tag>;
     }
 
-    if (status === "completed") {
-      return <Tag type="red">{t("completed", "Completed")}</Tag>;
+    if (
+      combinedStatus ===
+      MedicationRequestMedicationDispenseCombinedStatus.completed
+    ) {
+      return <Tag type="green">{t("completed", "Completed")}</Tag>;
     }
 
-    if (status === "expired") {
+    if (
+      combinedStatus ===
+      MedicationRequestMedicationDispenseCombinedStatus.expired
+    ) {
       return <Tag type="red">{t("expired", "Expired")}</Tag>;
     }
 
+    if (
+      combinedStatus ===
+      MedicationRequestMedicationDispenseCombinedStatus.declined
+    ) {
+      return <Tag type="red">{t("closed", "Closed")}</Tag>;
+    }
+
+    if (
+      combinedStatus ===
+      MedicationRequestMedicationDispenseCombinedStatus.on_hold
+    ) {
+      return <Tag type="red">{t("paused", "Paused")}</Tag>;
+    }
```

Comment:
To avoid confusion regarding labels and display text, we could standardize them to have the same wording

Reply:
I think you are referring to the different where the status is, say "on_hold" and the tag is "paused".  Generally, I agree it would be clearer if they were the same.  The issue is that I'm trying to stick with statuses as part of the FHIR specification (https://www.hl7.org/fhir/R4/valueset-medicationdispense-status.html), while the display text is coming from the user requirements from our BAs.    So I'm trying to stick to FHIR standards for the best behind-the-scenes modelling, but then deferring to the UI team what the display text should be. Does that answer your question? (And do you agree?)

Reply:
I see the conflict.All good.